bip-musig2-descriptors.mediawiki

```diff
@@ -0,0 +1,69 @@
+<pre>
+  BIP: musig2-descriptors
+  Layer: Applications
+  Title: musig() Descriptor Key Expression
+  Author: Ava Chow <me@achow101.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-musig2-descriptor
+  Status: Draft
+  Type: Informational
+  Created: 2024-01-15
+  License: BSD-2-Clause
+</pre>
+
+==Abstract==
+
+This document specifies a <tt>musig()</tt> key expression for output script descriptors.
+<tt>musig()</tt> expressions take multiple keys and produce an aggregate public key using BIP 327.
+
+==Copyright==
+
+This BIP is licensed under the BSD 2-clause license.
+
+==Motivation==
+
+BIP 327 introduces the MuSig2 Multi-Signature scheme. It is useful to have a way for keys to be used
+in a MuSig2 aggregate key to be expressed in descriptors so that wallets can more easily use MuSig2.
+
+==Specification==
+
+A new key expression is defined: <tt>musig()</tt>.
+
+===<tt>musig()</tt>===
+
+The <tt>musig(KEY, KEY, ..., KEY)</tt> expression can only be used inside of a <tt>tr()</tt>
+expression as a key expression. The aggregate public key is produced by using the <tt>KeyAgg</tt>
+algorithm on all KEYs specified in the expression after performing all specified derivation. As with
+script expressions, KEY can contain child derivation specified by <tt>/*</tt>. A new aggregate
+public key will be computed for each child index. Keys must be sorted with the <tt>KeySort</tt>
```

Comment:
Why KeySort when we already have a explicit ordering from the descriptor? It seems like an unnecessary complication.

Reply:
As I mentioned in https://github.com/bitcoin/bips/pull/1540#issuecomment-2061452468, there was a discussion about this and that was the conclusion. But I'm also having a hard time remembering the exact reason, maybe @jonasnick and/or @sipa can help me out here.

Reply:
After talking about this again with @sipa, the conclusion is that philosophically, MuSig2 operates over a set of keys, so the order should not matter. For serialization as an implementation detail, it does, so having it sorted is a logical order for an ostensibly unordered set.

Another argument for sorting is to make the recovery process easier,. As we know from multisig usage that exists currently, users are often surprised to learn that they had to know more than just their key and their cosigners. By sorting, we can eliminate this as a concern. Ostensibly, users should be backing up their descriptors, but in practice, especially with the prevalance of hardware signers which espouse that only the seed needs to be backed up, I don't think that's something that happens that often.